crates/solver/src/settlement_submission.rs

```diff
@@ -311,6 +313,20 @@ impl SolutionSubmitter {
         };
 
         let strategy_args = strategy.strategy_args().expect("unreachable code executed");
+
+        // No extra tip required if there is no revert risk
+        let (additional_tip_percentage_of_max_fee, max_additional_tip) =
+            if shared::gas_price_estimation::is_mainnet(&network_id)
+                && settlement.revertable() == Revertable::NoRisk
```

Comment:
Judging by the comment the condition should rather be `settlement.revertable() != Revertable::NoRisk`, right?
Also do we really have to add a special case for `mainnet` here?